sql/auto_install.sql

```diff
@@ -1,23 +1,90 @@
-  `content` longtext    COMMENT 'Mosaico content (JSON)' ,
-  `msg_tpl_id` int unsigned NULL COMMENT 'FK to civicrm_msg_template.',
-   PRIMARY KEY ( `id` ),
-   CONSTRAINT FK_civicrm_mosaico_template_msg_tpl_id FOREIGN KEY (`msg_tpl_id`) REFERENCES `civicrm_msg_template`(`id`) ON DELETE SET NULL
-)  ENGINE=InnoDB DEFAULT CHARACTER SET utf8 COLLATE utf8_unicode_ci  ;
+-- +--------------------------------------------------------------------+
+-- | Copyright CiviCRM LLC. All rights reserved.                        |
+-- |                                                                    |
+-- | This work is published under the GNU AGPLv3 license with some      |
+-- | permitted exceptions and without any warranty. For full license    |
+-- | and copyright information, see https://civicrm.org/licensing       |
+-- +--------------------------------------------------------------------+
+--
+-- Generated from schema.tpl
+-- DO NOT EDIT.  Generated by CRM_Core_CodeGen
+-- 
+
+-- +--------------------------------------------------------------------+
+-- | Copyright CiviCRM LLC. All rights reserved.                        |
+-- |                                                                    |
+-- | This work is published under the GNU AGPLv3 license with some      |
+-- | permitted exceptions and without any warranty. For full license    |
+-- | and copyright information, see https://civicrm.org/licensing       |
+-- +--------------------------------------------------------------------+
+--
+-- Generated from drop.tpl
+-- DO NOT EDIT.  Generated by CRM_Core_CodeGen
+--
+-- /*******************************************************
+-- *
+-- * Clean up the existing tables
+-- *
+-- *******************************************************/
+
+SET FOREIGN_KEY_CHECKS=0;
+
+DROP TABLE IF EXISTS `civicrm_mosaico_template`;
+DROP TABLE IF EXISTS `civicrm_mosaico_msg_template`;
+
+SET FOREIGN_KEY_CHECKS=1;
+-- /*******************************************************
+-- *
+-- * Create new tables
+-- *
+-- *******************************************************/
+
+-- /*******************************************************
+-- *
+-- * civicrm_mosaico_msg_template
+-- *
+-- * Mosaico Templates Table
+-- *
+-- *******************************************************/
+CREATE TABLE `civicrm_mosaico_msg_template` (
+
+
+     `id` int unsigned NOT NULL AUTO_INCREMENT  COMMENT 'Unique Settings ID',
+     `msg_tpl_id` int unsigned NOT NULL   COMMENT 'FK to civicrm_msg_template.',
+     `hash_key` varchar(32) NOT NULL   ,
+     `name` varchar(32) NOT NULL   COMMENT 'name',
+     `html` longtext NOT NULL   COMMENT 'HTML',
+     `metadata` longtext NOT NULL   COMMENT 'metadata',
+     `template` longtext NOT NULL   COMMENT 'template' 
+,
+        PRIMARY KEY (`id`)
+ 
+ 
+,          CONSTRAINT FK_civicrm_mosaico_msg_template_msg_tpl_id FOREIGN KEY (`msg_tpl_id`) REFERENCES `civicrm_msg_template`(`id`) ON DELETE CASCADE  
```

Comment:
I generally add `ENGINE=InnoDB` because otherwise I get quite a few complaints as apparently some shared servers are not setting that as default - including "civihosting" which seems particularly irresponsible but anyway..

Reply:
Added in now @mattwire